lib/dock.rb

```diff
@@ -0,0 +1,47 @@
+require './lib/boat'
+require './lib/renter'
+require 'pry'
+
+class Dock
+
+attr_reader :name, :max_rental_time, :log_hour, :boats, :renters
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @log_hour = 0
+    @boats = []
+    @renters = []
+  end
+
+  def rent(boat, renters)
+    @boats << boat
+    if @renters.include?(renters)
+    else
+    @renters << renters
+    end
+  end
+
+  def log_hour
+    @boats.each do |boat|
+      boat.add_hour
+    end
+  end
+
+  def return(boat)
```

Comment:
I think I see what you're trying to do here. You're trying to change the boat's hours_rented to 3 if it's greater than three. The problem is that this code is inside the Dock class, so it doesn't have access to the @hours_rented instance variable inside the Boat class. One thing you could do is add an attr_accessor inside your Boat class so that you could change it here.